x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/NameableClusterPrivilege.java

```diff
@@ -0,0 +1,19 @@
+/*
+ *
+ *  Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ *  or more contributor license agreements. Licensed under the Elastic License;
+ *  you may not use this file except in compliance with the Elastic License.
+ *
+ */
+
+package org.elasticsearch.xpack.core.security.authz.privilege;
+
+import org.elasticsearch.xpack.core.security.authz.RoleDescriptor;
+
+/**
+ * A {@link ClusterPrivilege} that has a name. The named cluster privileges can be referred simply by name within a
+ * {@link RoleDescriptor#getClusterPrivileges()}.
+ */
+public interface NameableClusterPrivilege extends ClusterPrivilege {
```

Comment:
I'm going to be pedantic here. It's not `Nameable`, it's `Named`.
_Nameable_ implies that you can give something a name, but the code that uses this class can't apply an artibrary _name_ to the privilege, so it isn't _nameable_.
Rather, we're talking about a privilege that _has a name_, that is, it is _named_.